vhost-device-scmi/src/vhu_scmi.rs

```diff
@@ -326,6 +347,68 @@ impl VuScmiBackend {
         debug!("Processing event queue finished");
         Ok(())
     }
+
+    fn notify_event_queue(&mut self, vring: &VringRwLock, device_event: u16) -> Result<()> {
+        if self.event_descriptors.is_empty() {
+            vring
+                .signal_used_queue()
+                .map_err(|_| VuScmiError::DescriptorNotFound)?;
+            return Ok(());
+        }
+
+        let desc_chain = &self.event_descriptors.pop().unwrap();
+        let descriptors: Vec<_> = desc_chain.clone().collect();
+
+        if descriptors.len() != 1 {
+            return Err(VuScmiError::UnexpectedDescriptorCount(descriptors.len()));
+        }
+        let desc = descriptors[0];
+        if !desc.is_write_only() {
+            return Err(VuScmiError::UnexpectedReadableDescriptor(0));
+        }
+        debug!(
+            "SCMI event response avail descriptor length: {}",
+            desc.len()
+        );
+        debug!("Calling SCMI notify");
+        if let Some(mut response) = self.scmi_handler.notify(device_event) {
+            let write_desc_len: usize = desc.len() as usize;
+            if response.len() > write_desc_len {
+                error!(
+                    "Response of length {} cannot fit into the descriptor size {}",
+                    response.len(),
+                    write_desc_len
+                );
+                response = response.communication_error();
+                if response.len() > write_desc_len {
+                    return Err(VuScmiError::InsufficientDescriptorSize(
+                        response.len(),
+                        write_desc_len,
+                    ));
+                }
+            }
+            desc_chain
+                .memory()
+                .write_slice(response.as_slice(), desc.addr())
+                .map_err(|_| VuScmiError::DescriptorWriteFailed)?;
+
+            if vring
+                .add_used(desc_chain.head_index(), response.len() as u32)
+                .is_err()
+            {
+                log::error!("Failed to send notification data to the ring");
+                return Err(VuScmiError::SendNotificationFailed);
+            }
+
+            let _ = vring
+                .signal_used_queue()
+                .map_err(|_| VuScmiError::SendNotificationFailed);
+        } else {
+            debug!("Get Notification Failed!");
+            return Ok(());
```

Comment:
we can remove this return (if we really want to return `Ok(())`)

Reply:
You are right, I will remove it.